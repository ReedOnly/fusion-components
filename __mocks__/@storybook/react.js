import renderer from 'react-test-renderer';
import addons, { mockChannel } from '@storybook/addons';
import { withFusionContext } from '../../.storybook/withFusionContext';

addons.setChannel(mockChannel());

export const action = (actionName) => jest.fn();
const fusionContext = withFusionContext();
export const storiesOf = (groupName) => {
    const api = {
        decorators: [],
        parameters: {},

        add(storyName, story) {
            if (api.parameters.skipTest) {
                return;
            }

            describe(groupName, () => {
                it(storyName, () => {
                    const component = renderer.create(fusionContext(story));

                    // @TODO Enable custom names once released:
                    // > https://github.com/facebook/jest/pull/2094
                    expect(component.toJSON())
                        .toMatchSnapshot
                        // `${groupName}.${storyName}`
                        ();
                });
            });

            return api;
        },

        addDecorator() {
            return api;
        },

        addParameters(parameters) {
            api.parameters = { ...api.parameters, ...parameters };
            return api;
        },
    };

    return api;
};

export const addDecorator = () => {};
