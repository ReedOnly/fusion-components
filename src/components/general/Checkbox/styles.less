@import "~styles/variables.less";
@import "~styles/colors.less";

.checkboxContainer {
  --checkbox-height: calc(var(--grid-unit) * 2px);
  --checkbox-width: calc(var(--grid-unit) * 2px);

  height: calc(var(--grid-unit) * 6px);
  padding: 0 calc((var(--grid-unit) * 2px) - 3px) 0
    calc((var(--grid-unit) * 2px) - 3px);
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  user-select: none;

  &.disabled {
    cursor: not-allowed;
  }

  &.comfortable {
    input + label {
      font-size: 14px;
    }
  }

  &.compact {
    input + label {
      font-size: 11px;
    }
  }

  input {
    opacity: 0;
    height: 0;
    width: 0;

    &:checked ~ .checkmark {
      &::after {
        display: block;
      }
    }

    &:focus ~ .checkmark {
      &::before {
        display: block;
      }
    }
  }

  &.primary {
    color: @color-primary;

    & > .checkmark {
      border-color: @color-primary;
    }

    & > input:checked ~ .checkmark {
      background-color: @color-primary;
    }

    & > input:checked:disabled ~ .checkmark {
      background-color: @color-primary-alt2;
    }

    & > input:disabled ~ .checkmark {
      border-color: @color-primary-alt2;
    }
  }

  &.secondary {
    color: @color-signal;

    & > .checkmark {
      border-color: @color-signal;
    }

    & > input:checked ~ .checkmark {
      background-color: @color-signal;
    }

    & > input:checked:disabled ~ .checkmark {
      background-color: @color-signal-alt2;
    }

    & > input:disabled ~ .checkmark {
      border-color: @color-signal-alt2;
    }
  }

  .checkmark {
    height: var(--checkbox-height);
    width: var(--checkbox-width);
    margin-right: calc(var(--grid-unit) * 1px);
    display: flex;
    justify-content: center;
    align-items: center;
    border: 2px solid;
    border-radius: 2px;
    box-sizing: border-box;

    &:after {
      width: calc(var(--checkbox-width) * 0.25);
      height: calc(var(--checkbox-height) * 0.6);
      margin-bottom: calc(var(--checkbox-height) * 0.13);
      border: solid white;
      border-width: 0 2px 2px 0;
      transform: rotate(45deg);
      content: "";
      display: none;
    }

    &:before {
      width: calc(var(--checkbox-width) + 2px);
      height: calc(var(--checkbox-height) + 2px);
      border: 1px dashed #bfbfbf;
      position: absolute;
      content: "";
      display: none;
    }
  }
}
