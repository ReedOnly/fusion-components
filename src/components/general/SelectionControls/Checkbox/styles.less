
@import '~styles/mixins.less';
@import '../inputStyles.less';


.checkboxContainer {
    .inputStyles(checkmark, var(--color-primary), var(--color-primary), var(--color-black-alt4));

    &.compact {
        input + label {
            &::before{
                width: .grid-unit(1.8px)[];
                height: .grid-unit(1.8px)[];
           
            }
        }
    }

    input + label {
        position: relative;
        &::before{
            border: 1px solid var(--color-primary);
            width: .grid-unit(2px)[];
            height: .grid-unit(2px)[];
            border-radius: 2px;
        }
      
        .checkmark {
            width: .grid-unit(2px)[];
            height: .grid-unit(2px)[];
            color: var(--color-white);
            transform: scale(0.8);
            opacity: 0;
            transition: all .1s;
        }
    }
  

    //Indeterminate
    input:indeterminate + label{
        &::before {
            border-color: var(--color-primary);
            background-color:  var(--color-primary);
        }

        .checkmark {
            transform: scale(1);
            opacity: 1;
        }
    }


}