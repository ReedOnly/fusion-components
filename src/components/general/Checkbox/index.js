import React, { Component } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import ComponentDisplayContext, {
  componentDisplayTypeClassNames
} from "../../contexts/ComponentDisplayContext";

import styles from "./styles.less";

const getCheckboxStyleClasses = props => ({
  [styles.primary]: props.primary,
  [styles.secondary]: props.secondary,
  [styles.disabled]: props.isDisabled
});

export default class Checkbox extends Component {
  static propTypes = {
    primary: PropTypes.bool,
    secondary: PropTypes.bool,
    onChange: PropTypes.func.isRequired,
    isChecked: PropTypes.bool,
    label: PropTypes.string,
    isDisabled: PropTypes.bool
  };

  static defaultProps = {
    primary: true,
    secondary: false,
    label: "",
    isChecked: false,
    isDisabled: false
  };

  renderCheckbox = displayType => {
    const { isChecked, isDisabled, onChange, label } = this.props;
    const commonClassName = {
      ...componentDisplayTypeClassNames(displayType, styles)
    };
    const checkboxClassName = classNames(
      styles.checkboxContainer,
      getCheckboxStyleClasses(this.props),
      commonClassName
    );

    return (
      <label
        className={checkboxClassName}
        disabled={isDisabled}
        onClick={e => e.stopPropagation()}
      >
        <input
          type="checkbox"
          checked={isChecked}
          disabled={isDisabled}
          onChange={onChange}
        />
        <span className={styles.checkmark} />
        {label}
      </label>
    );
  };

  render() {
    return (
      <ComponentDisplayContext.Consumer>
        {displayType => this.renderCheckbox(displayType)}
      </ComponentDisplayContext.Consumer>
    );
  }
}
