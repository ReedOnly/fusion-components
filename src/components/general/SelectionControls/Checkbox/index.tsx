import React, { useRef, MutableRefObject, forwardRef, useEffect } from 'react';
import styles from './styles.less';
import { DoneIcon, MinimizeIcon } from '@equinor/fusion-components';
import classNames from 'classnames';
import tinycolor from 'tinycolor2';
import { useColorOverrideFilter } from '../utils';

type CheckboxProps = {
    selected?: boolean;
    onChange?: (e: React.MouseEvent<HTMLDivElement>) => void;
    disabled?: boolean;
    indeterminate?: boolean;
    color?: string;
};

const Checkbox = forwardRef<HTMLInputElement | null, CheckboxProps>(
    ({ selected, onChange, disabled, indeterminate, color }, ref) => {
        const inputRef =
            (ref as MutableRefObject<HTMLInputElement | null>) || useRef<HTMLInputElement | null>();

        const containerClassNames = classNames(styles.checkboxContainer, {
            [styles.disabled]: disabled,
        });

        useEffect(() => {
            if (inputRef.current) {
                inputRef.current.indeterminate = indeterminate ? true : false;
            }
        }, [indeterminate]);

        const { hueFilter, slFilter } = useColorOverrideFilter(color);

        return (
            <div style={{ filter: hueFilter }} className={containerClassNames} onClick={onChange}>
                <input
                    type="checkbox"
                    checked={selected}
                    disabled={disabled}
                    readOnly
                    ref={inputRef}
                />
                <label style={{ filter: slFilter }}>
                    <span className={styles.checkmark}>
                        {indeterminate ? (
                            <MinimizeIcon width={16} height={16} />
                        ) : (
                            <DoneIcon width={16} height={16} />
                        )}
                    </span>
                </label>
            </div>
        );
    }
);

export default Checkbox;
