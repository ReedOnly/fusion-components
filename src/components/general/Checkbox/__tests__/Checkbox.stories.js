import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import { boolean, text } from "@storybook/addon-knobs";

import Checkbox from "../";

storiesOf("General components/Checkbox", module)
  .addParameters({ jest: ["Checkbox.stories"] })
  .add("default", () => (
    <div
      style={{
        display: "flex",
        flex: 1,
        justifyContent: "space-around",
        alignItems: "center"
      }}
    >
      <Checkbox
        primary={boolean("primary", true)}
        secondary={boolean("secondary", false)}
        isChecked={boolean("isChecked", false)}
        isDisabled={boolean("isDisabled", false)}
        label={text("label", "Label")}
        onChange={action("changed")}
      />
    </div>
  ));
