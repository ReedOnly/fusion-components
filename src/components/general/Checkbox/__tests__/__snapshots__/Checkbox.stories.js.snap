// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General components/Checkbox With label 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <div
    className="checkboxContainer comfortable"
  >
    <input
      checked={false}
      disabled={false}
      id="heisann"
      name="Heisann"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="heisann"
      onClick={[Function]}
    >
      <span
        className="checkboxLabel"
      >
        Heisann
      </span>
    </label>
  </div>
  <div
    className="checkboxContainer comfortable"
  >
    <input
      checked={false}
      disabled={true}
      id="heisann"
      name="Heisann"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="heisann"
      onClick={[Function]}
    >
      <span
        className="checkboxLabel"
      >
        Heisann
      </span>
    </label>
  </div>
  <div
    className="checkboxContainer comfortable"
  >
    <input
      checked={true}
      disabled={false}
      id="heisann"
      name="Heisann"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="heisann"
      onClick={[Function]}
    >
      <span
        className="checkboxLabel"
      >
        Heisann
      </span>
    </label>
  </div>
  <div
    className="checkboxContainer comfortable"
  >
    <input
      checked={true}
      disabled={true}
      id="heisann"
      name="Heisann"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="heisann"
      onClick={[Function]}
    >
      <span
        className="checkboxLabel"
      >
        Heisann
      </span>
    </label>
  </div>
</div>
`;

exports[`General components/Checkbox Without label 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <div
    className="checkboxContainer comfortable"
  >
    <input
      checked={false}
      disabled={false}
      id=""
      name=""
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor=""
      onClick={[Function]}
    >
      <span
        className="checkboxLabel"
      >
        
      </span>
    </label>
  </div>
  <div
    className="checkboxContainer comfortable"
  >
    <input
      checked={false}
      disabled={true}
      id=""
      name=""
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor=""
      onClick={[Function]}
    >
      <span
        className="checkboxLabel"
      >
        
      </span>
    </label>
  </div>
  <div
    className="checkboxContainer comfortable"
  >
    <input
      checked={true}
      disabled={false}
      id=""
      name=""
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor=""
      onClick={[Function]}
    >
      <span
        className="checkboxLabel"
      >
        
      </span>
    </label>
  </div>
  <div
    className="checkboxContainer comfortable"
  >
    <input
      checked={true}
      disabled={true}
      id=""
      name=""
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor=""
      onClick={[Function]}
    >
      <span
        className="checkboxLabel"
      >
        
      </span>
    </label>
  </div>
</div>
`;
